pkg/network/netlink/consumer.go

```diff
@@ -128,19 +132,143 @@ func (c *Consumer) Events() <-chan Event {
 		defer close(output)
 		c.streaming = true
 		_ = c.conn.JoinGroup(netlinkCtNew)
-		c.receive(output)
+		c.receive(output, c.socket)
 	})
 
 	return output
 }
 
+func isPeerNS(conn *netlink.Conn, ns netns.NsHandle) bool {
+	encoder := netlink.NewAttributeEncoder()
+	encoder.Uint32(unix.NETNSA_FD, uint32(ns))
+	data, err := encoder.Encode()
+	if err != nil {
+		log.Tracef("isPeerNS: err encoding attributes netlink attributes: %s", err)
+		return false
+	}
+
+	msg := netlink.Message{
+		Header: netlink.Header{
+			Flags: netlink.Request | netlink.Acknowledge,
+			Type: unix.RTM_GETNSID,
+			Sequence: netlinkSeqNumber,
+		},
+		Data: []byte{unix.AF_UNSPEC, 0, 0, 0},
+	}
+
+	msg.Data = append(msg.Data, data...)
+
+	if msg, err = conn.Send(msg); err != nil {
+		log.Tracef("isPeerNS: err sending netlink request: %s", err)
```

Comment:
Why are these logged as trace and not something higher?

Reply:
Don't know why I chose `Tracef`, these should be `Errorf`. Fixed